Session1/EFCoreAssignment/Program.cs

```diff
@@ -0,0 +1,67 @@
+﻿// See https://aka.ms/new-console-template for more information
+using EFCoreAssignment;
+using Microsoft.EntityFrameworkCore;
+
+Console.WriteLine("Hello, World!");
+
+var connection = @"Server=.\SQLEXPRESS;Database=Frederick.EFCoreMasters.Session01;Trusted_Connection=True"; // TODO : Add your connection string here
+var optionsBuilder =
+    new DbContextOptionsBuilder
+           <AppDbContext>();
+optionsBuilder.UseSqlServer(connection);
+
+var options = optionsBuilder.Options;
+
+var dbContext = new AppDbContext(options);
+
+Filtering(dbContext);
+SingleOrDefault(dbContext);
+LoadingRelatedData_Manual(dbContext);
+LoadingRelatedData_ExplicitLoading(dbContext);
+LoadingRelatedData_EagerLoading(dbContext);
+
+static void Filtering(AppDbContext dbContext)
+{
+    // TODO : Filter by Product Name
+    var products = dbContext.Products.AsNoTracking().Where(product => product.Name == "Product A");
+
+    foreach (var product in products)
+    {
+        Console.WriteLine($"Filtering - Product: {product.Name}");
+    }
+}
+
+static void SingleOrDefault(AppDbContext dbContext)
+{
+    // TODO : Select using SingleOrDefault by Product Id   
+    var product = dbContext.Products.AsNoTracking().SingleOrDefault(product => product.Id== 2);
+    Console.WriteLine($"SingleOrDefault - Product: {product!.Name}");
+}
+
+static void LoadingRelatedData_Manual(AppDbContext dbContext)
+{
+    // TODO : Load Product with related shop data manually
+    var product = dbContext.Products.AsNoTracking().FirstOrDefault(product => product.Id == 3);
+    product!.Shop = dbContext.Shops.Single(shop => shop.Id == product.ShopId);
```

Comment:
This is fine, but It'll be better to add validation to check if the product is null before loading the related data